### PR TITLE
Allow overwriting existing output file

### PR DESCRIPTION
**Pull request by romanz**
Otherwise, `rage` fails with:
```bash 
$ echo | rage -r age100qf7fggvctqclsqvl2xu6vjnrerxm9vt8x56yql6gaz9x54vcaqdh2q26 -o existing-file.age
Error: File exists (os error 17)
```

### CODE REVIEW

**Comment by romanz** [comment]
IIUC, all the tests pass except uploading benchmarks - which fails with:
```
Must provide CRITERION_TOKEN in environment: export CRITERION_TOKEN="token"
Error: Process completed with exit code 1.
```
https://github.com/str4d/rage/pull/168/checks?check_run_id=1708942847

**Comment by romanz** [comment]
Ping :)

File: age/src/cli_common/file_io.rs [hunk 1]
```diff
@@ -270,3 +270,18 @@ impl Write for OutputWriter {
         }
     }
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::{OutputFormat, OutputWriter};
+    use std::io::Write;
```
Comment (str4d) [inline age/src/cli_common/file_io.rs:277]:
```suggestion
    #[cfg(unix)]
    use super::{OutputFormat, OutputWriter};
    #[cfg(unix)]
    use std::io::Write;
```

**Review by str4d — APPROVED** [review]
Thanks for the PR! I've been reluctant to merge it because I'm not personally a fan of just overwriting files, but after talking with @FiloSottile and some other sysadmins, I now agree that the more consistent UX is for `-o` to behave the same way as pipes, i.e. do what Unix does and overwrite the output.

**Comment by romanz** [comment]
Thanks!

**Pull request merged**